src/VecSim/spaces/computer/preprocessors.h

```diff
@@ -57,9 +57,11 @@ class CosinePreprocessor : public PreprocessorInterface {
             // If one of them is null, allocate memory for it and copy the original_blob to it.
             if (storage_blob == nullptr) {
                 storage_blob = this->allocator->allocate(processed_bytes_count);
+                // TODO: handle original_blob_size != processed_bytes_count
```

Comment:
When? Is there something that will fail a PR if we forget to support it? (relevant to all the changes in the file)